Project/Source/Components/Physics/ComponentBoxCollider.cpp

```diff
@@ -167,11 +167,11 @@ void ComponentBoxCollider::Load(JsonValue jComponent) {
 	freezeRotation = jFreeze;
 }
 
-void ComponentBoxCollider::OnCollision() {
+void ComponentBoxCollider::OnCollision(const GameObject& collidedWith) {
```

Comment:
Why is this const? We may want to edit the other object's position or something in response.

Reply:
But that will be done in that objects' OnCollision, don't you think? This collidedWith should be only used to access properties of the other object in my opinion (e.g. the enemy OnCollision needs to know the damage of the bullet that collided with him, but the deletion of the bullet will be performed inside the OnCollision method of the bullet)